### PR TITLE
CP-38473: Block RPU if the pool has hosts with certificate key length less than 2048.

### PR DESCRIPTION
**Pull request by kc284**

### CODE REVIEW

File: XenAdmin/Diagnostics/Checks/CertificateKeyLengthCheck.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            return null;
+        }
+
+        private bool IsShortCertificateKey()
+        {
+            var session = Host.Connection?.Session;
+            if (session == null)
+                return false;
+
+            var certificate = Host.get_server_certificate(session, Host.opaque_ref);
```
Comment (minglumlu) [inline XenAdmin/Diagnostics/Checks/CertificateKeyLengthCheck.cs:93]: Would it be possible that this certificate is a chain? If so, will the `X509Certificate2` get the correct result? (I'm not familiar with the C# APIs for this.)

Comment (kc284) [reply XenAdmin/Diagnostics/Checks/CertificateKeyLengthCheck.cs:93]: I checked with Toolstack, they said the call returns a single certificate.

File: XenAdmin/Diagnostics/Checks/CertificateKeyLengthCheck.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            : base(coordinator)
+        {
+            _manualUpgrade = manualUpgrade;
+            _installMethodConfig = installMethodConfig;
+        }
+
+        protected override Problem RunHostCheck()
+        {
+            //post-stockholm uses stunnel 5.60 which requires certificate key length
+            //greater than 2048 bytes; yangtze uses stunnel 5.56
```
Comment (AaronRobson) [inline XenAdmin/Diagnostics/Checks/CertificateKeyLengthCheck.cs:59]: Greater than *or equal to* 2048 bytes?

Comment (kc284) [reply XenAdmin/Diagnostics/Checks/CertificateKeyLengthCheck.cs:59]: Indeed it's >=2048; I'll correct the comment.

**Review by AaronRobson — APPROVED** [review]

**Review by CitrixChris — APPROVED** [review]

**Review by CitrixChris — CHANGES_REQUESTED** [review]
Will approve once Aaron's comment is resolved

**Review by CitrixChris — APPROVED** [review]

**Pull request merged**